Classes/common/touchdb/TD_Database.m

```diff
@@ -438,24 +577,43 @@ - (BOOL)close
     _activeReplicators = nil;
 
     [_fmdbQueue close];
+    _fmdbQueue = nil;
+    _keyProviderToOpenDB = nil;
 
     _open = NO;
     _transactionLevel = 0;
+    
     return YES;
 }
 
-- (BOOL)deleteDatabase:(NSError**)outError
+- (BOOL)deleteDatabase:(NSError **)outError
 {
-    CDTLogInfo(CDTDATASTORE_LOG_CONTEXT, @"Deleting %@", _path);
     [[NSNotificationCenter defaultCenter] postNotificationName:TD_DatabaseWillBeDeletedNotification
                                                         object:self];
-    if (_open) {
-        if (![self close]) return NO;
-    } else if (!self.exists) {
-        return YES;
+
+    if (_open && ![self close]) {
```

Comment:
As a resume, the main concern here is that this code is not thread-safe. However, we have this problem not only when we 'delete' a db, we also have the same issue when we 'open' and 'close' a db.

Once a db is created and opened, all operations are executed in a serial FMDBQueue but the operations mentioned before are executed in the same thread they are called and there are not locks in place to manage concurrency.

To end, this is a bug and we will address it in a new PR.